src/consumption.cpp

```diff
@@ -540,7 +540,11 @@ int Character::vitamin_mod( const vitamin_id &vit, int qty, bool capped )
 {
     auto it = vitamin_levels.find( vit );
     if( it == vitamin_levels.end() ) {
-        return 0;
+        if( !vit.is_valid() ) {
+            debugmsg( "Vitamin with id %s does not exist, and cannot be modified", vit.str() );
+            return 0;
+        }
+        it = vitamin_levels.emplace( vit, 0 ).first;
     }
```

Comment:
No need to `find` and then `insert`; `insert` doesn't modify if it's already there so this whole section can just be
```
auto it = vitamin_levels.emplace( vit, 0 ).first;
```

Reply:
Feels a little dirty, but it's done.